contracts/Remittance.sol

```diff
@@ -1,38 +1,40 @@
 pragma solidity >=0.4.21 <0.6.0;
 
 import "./Pausable.sol";
-import "./Ownable.sol";
 
 contract Remittance is Pausable {
-  address public carol;
-  bytes32 private withdrawHash;
+  struct Escrow {
+    address carol;
+    uint256 amount;
+  }
+
+  mapping (bytes32 => Escrow) public escrows;
 
   event LogDeposited(address indexed sender, uint256 amount, bytes32 hash, address indexed carol);
-  event LogWithdrawn(address indexed sender, uint256 amount);
+  event LogWithdrawn(address indexed sender, bytes32 hash, uint256 amount);
 
   constructor(bool paused) Pausable(paused) public {
   }
 
-  function generateHash(address _carol, bytes32 _bobPassword, bytes32 _carolPassword) public view returns(bytes32) {
-     return keccak256(abi.encodePacked(this, _carol, _bobPassword, _carolPassword));
+  function generateHash(address carol, bytes32 bobPassword, bytes32 carolPassword) public view returns(bytes32) {
+     return keccak256(abi.encodePacked(this, carol, bobPassword, carolPassword));
   }
 
-  function deposit(bytes32 _hash, address _carol) external payable onlyOwner whenRunning whenAlive {
-    require(_carol != address(0), "Carol's address must not be zero!");
+  function deposit(bytes32 hash, address carol) external payable onlyOwner whenRunning whenAlive {
+    require(carol != address(0), "Carol's address must not be zero!");
     require(msg.value > 0, "The value of deposit must be more than zero ether!");
-    carol = _carol;
-    withdrawHash = _hash;
-    emit LogDeposited(msg.sender, msg.value, _hash, _carol);
+    escrows[hash] = Escrow(carol, msg.value);
+    emit LogDeposited(msg.sender, msg.value, hash, carol);
   }
 
-  function withdraw(bytes32 _bobPassword, bytes32 _carolPassword) external whenRunning whenAlive {
-    require(msg.sender == carol, "Only Carol can withdraw!");
-    uint256 contractBalance = address(this).balance;
-    require(contractBalance > 0, "Can't withdraw since the contract balance is zero!");
+  function withdraw(bytes32 bobPassword, bytes32 carolPassword) external whenRunning whenAlive {
+    bytes32 hash = generateHash(msg.sender, bobPassword, carolPassword);
```

Comment:
Can you think about `carol` in the hash, `carol` in the `Escrow` and `carolPassword`, and tell me if there is not something that can be done here?